hydra-node/exe/hydra-node/Main.hs

```diff
@@ -39,13 +40,17 @@ main = do
       let headState = createHeadState [] (HeadParameters 3 mempty)
       hh <- createHydraHead headState Ledger.mockLedger
       oc <- createMockChainClient eq (contramap MockChain tracer)
-      withHeartbeat nodeId (withOuroborosNetwork (show host, port) peers) (putEvent eq . NetworkEvent) $ \hn -> do
-        responseChannel <- newBroadcastTChanIO
-        let sendResponse = atomically . writeTChan responseChannel
-        let node = HydraNode{eq, hn, hh, oc, sendResponse, env}
-        race_
-          (runAPIServer apiHost apiPort responseChannel node (contramap APIServer tracer))
-          (runHydraNode (contramap Node tracer) node)
+      withHeartbeat
+        nodeId
+        (withBroadcastToSelf $ withOuroborosNetwork (show host, port) peers)
```

Comment:
How about a `withNetwork nodeId host peers = withHeartbeat nodeId (withBroadcastToSelf $ withOuroborosNetwork (show host, port) peers)`? IMO that would really drive home the composability argument for this technique :)